riscv/csrs.cc

```diff
@@ -503,6 +515,11 @@ mstatus_csr_t::mstatus_csr_t(processor_t* const proc, const reg_t addr):
   ) {
 }
 
+reg_t mstatus_csr_t::read() const noexcept {
+  return val 
+         | adjust_sd(set_field((reg_t)0, SSTATUS_XS, proc->get_xs_gatherer()->get_xs()));
```

Comment:
You may be right that this causes a significant performance loss. If so, optimization should come as a future commit or PR.

Reply:
Thanks, I will optimize this in another PR after this one is merged.